src/core/shape_inference/include/rdft_shape_inference.hpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/rdft.hpp>
+
+#include "openvino/core/axis_vector.hpp"
+#include "utils.hpp"
+
+template <typename B>
+B get_ouput_dimension_bound(B b) {
+    if (b <= 0) {
+        return b;
+    }
```

Comment:
Is it valid to have negative output dimension? Maybe we should throw exception here?

Reply:
It is not a negative _dimension_, but it is a negative _bound of a dimension_. Such bound can be exists when the dimension is dynamic.